[grpc] build failure macOS arm64

### Operating system

macOS

### Compiler

clang 

### Steps to reproduce the behavior

```Shell
./vcpkg install grpc:arm64-osx-dynamic-release
```
with triplets details

```
set(VCPKG_BUILD_TYPE release)
set(VCPKG_TARGET_ARCHITECTURE arm64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)

set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
set(VCPKG_OSX_ARCHITECTURES arm64)
set(VCPKG_OSX_DEPLOYMENT_TARGET 12.0)
set(CMAKE_OSX_DEPLOYMENT_TARGET 12.0)
```

`
 /Applications/Xcode-16.2.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -fPIC  -O3 -DNDEBUG -arch arm64 -isysroot /Applications/Xcode-16.2.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=12.0 -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 46.0.0 -current_version 46.0.0 -o libupb_json_lib.46.0.0.dylib -install_name @rpath/libupb_json_lib.46.dylib CMakeFiles/upb_json_lib.dir/third_party/upb/upb/json/decode.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/json/encode.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/lex/atoi.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/lex/round_trip.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/lex/strtod.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/lex/unicode.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/message/copy.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/message/merge.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/def_pool.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/def_type.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/desc_state.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/enum_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/enum_reserved_range.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/enum_value_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/extension_range.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/field_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/file_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/internal/def_builder.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/internal/strdup2.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/message.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/message_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/message_reserved_range.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/method_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/oneof_def.c.o CMakeFiles/upb_json_lib.dir/third_party/upb/upb/reflection/service_def.c.o  -Wl,-rpath,/opt/proj/libs/vcpkg/buildtrees/grpc/arm64-osx-dynamic-release-rel  -lm  libupb_mini_descriptor_lib.46.0.0.dylib  libupb_wire_lib.46.0.0.dylib  /opt/proj/libs/vcpkg/installed/arm64-osx-dynamic-release/lib/libutf8_range.a  libupb_message_lib.46.0.0.dylib  libupb_base_lib.46.0.0.dylib  libupb_mem_lib.46.0.0.dylib  -lm && :
ld: warning: ignoring duplicate libraries: '-lm'
Undefined symbols for architecture arm64:
  "_google__protobuf__DescriptorProto__ExtensionRange_msg_init", referenced from:
      __upb_MessageDefs_New in message_def.c.o
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__DescriptorProto__ReservedRange_msg_init", referenced from:
      __upb_MessageDefs_New in message_def.c.o
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__DescriptorProto_msg_init", referenced from:
      __upb_FileDef_Create in file_def.c.o
      __upb_FileDef_Create in file_def.c.o
      _count_exts_in_msg in file_def.c.o
      _count_exts_in_msg in file_def.c.o
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__EnumDescriptorProto__EnumReservedRange_msg_init", referenced from:
      __upb_EnumDefs_New in enum_def.c.o
  "_google__protobuf__EnumDescriptorProto_msg_init", referenced from:
      __upb_FileDef_Create in file_def.c.o
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__EnumOptions_msg_init", referenced from:
      __upb_EnumDefs_New in enum_def.c.o
      __upb_EnumDefs_New in enum_def.c.o
      __upb_EnumDefs_New in enum_def.c.o
  "_google__protobuf__EnumValueDescriptorProto_msg_init", referenced from:
      __upb_EnumDefs_New in enum_def.c.o
  "_google__protobuf__EnumValueOptions_msg_init", referenced from:
      __upb_EnumValueDefs_New in enum_value_def.c.o
      __upb_EnumValueDefs_New in enum_value_def.c.o
      __upb_EnumValueDefs_New in enum_value_def.c.o
  "_google__protobuf__ExtensionRangeOptions_msg_init", referenced from:
      __upb_ExtensionRanges_New in extension_range.c.o
      __upb_ExtensionRanges_New in extension_range.c.o
      __upb_ExtensionRanges_New in extension_range.c.o
  "_google__protobuf__FeatureSetDefaults__FeatureSetEditionDefault_msg_init", referenced from:
      _upb_DefPool_SetFeatureSetDefaults in def_pool.c.o
      __upb_FileDef_FindEdition in file_def.c.o
  "_google__protobuf__FeatureSetDefaults_msg_init", referenced from:
      _upb_DefPool_SetFeatureSetDefaults in def_pool.c.o
      _upb_DefPool_SetFeatureSetDefaults in def_pool.c.o
  "_google__protobuf__FeatureSet_msg_init", referenced from:
      _upb_DefBuilder_AddFileToPool in def_pool.c.o
      __upb_EnumDefs_New in enum_def.c.o
      __upb_EnumValueDefs_New in enum_value_def.c.o
      __upb_ExtensionRanges_New in extension_range.c.o
      __upb_FieldDef_Create in field_def.c.o
      __upb_FileDef_FindEdition in file_def.c.o
      __upb_FileDef_Create in file_def.c.o
      ...
  "_google__protobuf__FieldDescriptorProto_msg_init", referenced from:
      __upb_FileDef_Create in file_def.c.o
      __upb_FileDef_Create in file_def.c.o
      _count_exts_in_msg in file_def.c.o
      __upb_MessageDefs_New in message_def.c.o
      __upb_MessageDefs_New in message_def.c.o
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__FieldOptions_msg_init", referenced from:
      __upb_FieldDef_Create in field_def.c.o
      __upb_FieldDef_Create in field_def.c.o
  "_google__protobuf__FileDescriptorProto_msg_init", referenced from:
      __upb_DefPool_LoadDefInitEx in def_pool.c.o
      __upb_DefPool_LoadDefInitEx in def_pool.c.o
  "_google__protobuf__FileOptions_msg_init", referenced from:
      __upb_FileDef_Create in file_def.c.o
      __upb_FileDef_Create in file_def.c.o
  "_google__protobuf__MessageOptions_msg_init", referenced from:
      __upb_MessageDefs_New in message_def.c.o
      __upb_MessageDefs_New in message_def.c.o
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__MethodDescriptorProto_msg_init", referenced from:
      __upb_ServiceDefs_New in service_def.c.o
  "_google__protobuf__MethodOptions_msg_init", referenced from:
      __upb_MethodDefs_New in method_def.c.o
      __upb_MethodDefs_New in method_def.c.o
      __upb_MethodDefs_New in method_def.c.o
  "_google__protobuf__OneofDescriptorProto_msg_init", referenced from:
      __upb_MessageDefs_New in message_def.c.o
  "_google__protobuf__OneofOptions_msg_init", referenced from:
      __upb_OneofDefs_New in oneof_def.c.o
      __upb_OneofDefs_New in oneof_def.c.o
      __upb_OneofDefs_New in oneof_def.c.o
  "_google__protobuf__ServiceDescriptorProto_msg_init", referenced from:
      __upb_FileDef_Create in file_def.c.o
  "_google__protobuf__ServiceOptions_msg_init", referenced from:
      __upb_ServiceDefs_New in service_def.c.o
      __upb_ServiceDefs_New in service_def.c.o
      __upb_ServiceDefs_New in service_def.c.o
ld: symbol(s) not found for architecture arm64
cc: error: linker command failed with exit code 1 (use -v to see invocation)`

### Additional context

similar to [this](https://github.com/grpc/grpc/issues/36654) ,  (https://github.com/grpc/grpc/issues/39170)

--- a/ports/grpc/00017-add-src-upb.patch
+++ b/ports/grpc/00017-add-src-upb.patch
@@ -1,0 +1,36 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 391d113..3774159 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -3969,6 +3969,7 @@ add_library(upb_json_lib ${_gRPC_STATIC_WIN32}
+   third_party/upb/upb/reflection/method_def.c
+   third_party/upb/upb/reflection/oneof_def.c
+   third_party/upb/upb/reflection/service_def.c
++  third_party/upb/upb/reflection/cmake/google/protobuf/descriptor.upb_minitable.c
+ )
+ 
+ target_compile_features(upb_json_lib PUBLIC cxx_std_17)
+@@ -3992,7 +3993,6 @@ endif()
+ target_include_directories(upb_json_lib
+   PUBLIC $<INSTALL_INTERFACE:${gRPC_INSTALL_INCLUDEDIR}> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+   PRIVATE
+-    ${CMAKE_CURRENT_SOURCE_DIR}
+     ${_gRPC_ADDRESS_SORTING_INCLUDE_DIR}
+     ${_gRPC_RE2_INCLUDE_DIR}
+     ${_gRPC_SSL_INCLUDE_DIR}
+@@ -4226,6 +4226,7 @@ add_library(upb_textformat_lib ${_gRPC_STATIC_WIN32}
+   third_party/upb/upb/reflection/service_def.c
+   third_party/upb/upb/text/encode.c
+   third_party/upb/upb/text/internal/encode.c
++  third_party/upb/upb/reflection/cmake/google/protobuf/descriptor.upb_minitable.c
+ )
+ 
+ target_compile_features(upb_textformat_lib PUBLIC cxx_std_17)
+@@ -4249,7 +4250,6 @@ endif()
+ target_include_directories(upb_textformat_lib
+   PUBLIC $<INSTALL_INTERFACE:${gRPC_INSTALL_INCLUDEDIR}> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+   PRIVATE
+-    ${CMAKE_CURRENT_SOURCE_DIR}
+     ${_gRPC_ADDRESS_SORTING_INCLUDE_DIR}
+     ${_gRPC_RE2_INCLUDE_DIR}
+     ${_gRPC_SSL_INCLUDE_DIR}

--- a/ports/grpc/portfile.cmake
+++ b/ports/grpc/portfile.cmake
@@ -16,6 +16,7 @@ vcpkg_from_github(
         00006-utf8-range.patch
         00015-disable-download-archive.patch
         00016-fix-plugin-targets.patch
+        00017-add-src-upb.patch
 )
 # Ensure de-vendoring
 file(REMOVE_RECURSE

--- a/ports/grpc/vcpkg.json
+++ b/ports/grpc/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "grpc",
   "version-semver": "1.71.0",
+  "port-version": 1,
   "description": "gRPC is a modern, open source, high-performance remote procedure call (RPC) framework that can run anywhere. gRPC enables client and server applications to communicate transparently, and simplifies the building of connected systems.",
   "homepage": "https://github.com/grpc/grpc",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3354,7 +3354,7 @@
     },
     "grpc": {
       "baseline": "1.71.0",
-      "port-version": 0
+      "port-version": 1
     },
     "grppi": {
       "baseline": "0.4.0",

--- a/versions/g-/grpc.json
+++ b/versions/g-/grpc.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "6c5319c24f882d2b64f517a1c00cac89d166f063",
+      "git-tree": "27a155298ad082983b805ed24043114fa0ca56af",
       "version-semver": "1.71.0",
       "port-version": 0
     },

--- a/versions/g-/grpc.json
+++ b/versions/g-/grpc.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "fe12b2f1ba6da9373174a6954aec7038f4278a2e",
+      "version-semver": "1.71.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "27a155298ad082983b805ed24043114fa0ca56af",
       "version-semver": "1.71.0",
       "port-version": 0

--- a/versions/g-/grpc.json
+++ b/versions/g-/grpc.json
@@ -6,7 +6,7 @@
       "port-version": 1
     },
     {
-      "git-tree": "27a155298ad082983b805ed24043114fa0ca56af",
+      "git-tree": "6c5319c24f882d2b64f517a1c00cac89d166f063",
       "version-semver": "1.71.0",
       "port-version": 0
     },

